src/ioctls/vcpu.rs

```diff
@@ -19,6 +19,15 @@ use vmm_sys_util::ioctl::{ioctl, ioctl_with_mut_ref, ioctl_with_ref};
 #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
 use vmm_sys_util::ioctl::{ioctl_with_mut_ptr, ioctl_with_ptr, ioctl_with_val};
 
+/// Guard trait for types that can hold arm register values
+/// SAFETY: Type that implements this trait must ensure that it is valid to initialize it
+/// by writing to it throught the pointer and it is big enough to hold register value
+pub unsafe trait ArmRegister: Sized {}
+
+unsafe impl ArmRegister for u32 {}
+unsafe impl ArmRegister for u64 {}
+unsafe impl ArmRegister for u128 {}
```

Comment:
Do we also want to implement this for `[u128; N]` for reasonable values of `N` (e.g. `128 * N <= 2048`) or something?

Reply:
This would implement it for types we don't want like `u384` since the max is fairly constricted could it be
```
unsafe impl ArmRegister for [u8;1] {} // 8
unsafe impl ArmRegister for [u8;2] {} // 16
unsafe impl ArmRegister for [u8;4] {} // 32
unsafe impl ArmRegister for [u8;8] {} // 64
unsafe impl ArmRegister for [u8;16] {} // 128
unsafe impl ArmRegister for [u8;32] {} // 256
unsafe impl ArmRegister for [u8;64] {} // 512
unsafe impl ArmRegister for [u8;128] {} // 1024
unsafe impl ArmRegister for [u8;256] {} // 2048
```

Reply:
@JonathanWoollett-Light I have already switched to `buffer` approach from above.